### ISSUE TITLE
_expand_algs fails with TypeError: can only concatenate tuple (not "list") to tuple

### ISSUE THREAD

**Issue by lexh147** [ronf/asyncssh#406]
Hello,

This simple script:
``` python
import asyncssh
options = asyncssh.SSHClientConnectionOptions()
```

produces the following traceback on 2.7.1, but not on 2.7.0:
```
Traceback (most recent call last):
  File "/Users/ahonore/src/lm/standalone/bug.py", line 3, in <module>
    options = asyncssh.SSHClientConnectionOptions()
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/connection.py", line 5595, in __init__
    super().__init__(options=options, last_config=last_config, **kwargs)
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/misc.py", line 268, in __init__
    self.prepare(**self.kwargs)
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/connection.py", line 6212, in prepare
    _select_host_key_algs(server_host_key_algs,
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/connection.py", line 454, in _select_host_key_algs
    return _select_algs('host key', algs, possible_algs,
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/connection.py", line 426, in _select_algs
    algs = _expand_algs(alg_type, algs, possible_algs,
  File "/Users/ahonore/miniconda3/envs/lm/lib/python3.9/site-packages/asyncssh/connection.py", line 405, in _expand_algs
    return default_algs + matched
TypeError: can only concatenate tuple (not "list") to tuple
```

Could it be due to https://github.com/ronf/asyncssh/commit/90d75903aacf280d89de8e87cfb853cc29225878 ?

Regards,
Alex H

**Comment by ronf** [comment]
Thanks for the report, and sorry you ran into this!

At first, I wasn't able to reproduce the problem. However, after a closer look, it seems to be triggered only when you have a line like the following in your .ssh/config file which tried to extend the list of supported algorithms with a leading '+':

```
HostKeyAlgorithms +ssh-dss
```

It can also be triggered by passing in an args argument with a leading '+':

```python
options = asyncssh.SSHClientConnectionOptions(server_host_key_algs='+ssh-dss')
```

As you said, it seems to be related to the removal of the following lines from `_select_host_key_algs()` in 2.7.1:

```python
    if not default_algs:
        default_algs = (get_default_certificate_algs() +
                        get_default_public_key_algs())
```

I removed this code to avoid duplication but missed that allowing default_args to be left at its default of () instead of a list could fail later on in _expand_algs. Bringing this code back solves the problem, but here's a better fix on top of 2.7.1 which doesn't require that:

```patch
--- /tmp/asyncssh-2.7.1/asyncssh/connection.py	2021-09-06 12:19:02.000000000 -0700
+++ /tmp/asyncssh-tmp/asyncssh/connection.py	2021-09-09 06:39:43.000000000 -0700
@@ -445,7 +445,7 @@
         raise ValueError('No %s algorithms selected' % alg_type)


-def _select_host_key_algs(algs, config_algs, default_algs=()):
+def _select_host_key_algs(algs, config_algs, default_algs):
     """Select a set of allowed host key algorithms"""

     possible_algs = (get_x509_certificate_algs() + get_certificate_algs() +
@@ -6210,7 +6210,7 @@
         # Just validate the input here -- the actual server host key
         # selection is done later, after the known_hosts lookup is done.
         _select_host_key_algs(server_host_key_algs,
-                              config.get('HostKeyAlgorithms', ()))
+                              config.get('HostKeyAlgorithms', ()), [])

         if username == ():
             username = config.get('User', local_username)
```

**Comment by lexh147** [comment]
Thank you Ron for the quick response ! Indeed I do have `HostKeyAlgorithms=+ssh-dss` in my SSH `config` file. For now we'll just pin to `2.7.0` so that our users do not need to apply a patch. Cheers, Alex

**Comment by ronf** [comment]
This fix is now available as commit f019fad in the "develop" branch and will be rolled into the next release.

**Comment by lexh147** [comment]
Thank you for the fix !

**Comment by ronf** [comment]
This fix is now released in AsyncSSH 2.7.2. Please let me know if you are seeing any issues with the new version!

**Comment by lexh147** [comment]
Works ! Thank you !
